File: cpp/problem350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> mp1, mp2;
        vector<int> res;
        for(auto i: nums1) mp1[i] ++;
        for(auto i: nums2) mp2[i] ++;
        for(const auto &pr: nums2){
            auto temp = mp1.find(pr.first);
            if( temp != mp1.cend() )
                fill_n( back_inserter(res), min( pr.second, temp->second ), pr.first );
        }
        return res;
    }
};

File: python/problem350.py
class Solution:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        se=set(nums1)&set(nums2)
        dic={}
        for i in se:
            dic[i]=min(nums1.count(i),nums2.count(i))
        li=[]
        for i in dic:
            for k in range(dic[i]):
                li.append(i)
        return li      
        
# from  https://leetcode.com/problems/intersection-of-two-arrays-ii/discuss/195266/Simple-python-solution-beats-99-with-explanation-(not-a-fancy-2-liner)      
class Solution2:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res = []
        map = {}
        for x in nums1:
        if x not in map:
            map[x] = 1
        else:
            map[x] += 1

        for y in nums2:
        if y in map:
            res.append(y)
            map[y] -=1
            if map[y] == 0:
                del(map[y])
        return res
        
# from  https://leetcode.com/problems/intersection-of-two-arrays-ii/discuss/191763/python-one-line-solve-this-problem       
class Solution3:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
       return list((collections.Counter(nums1) & collections.Counter(nums2)).elements())
        
        
        
       